### ISSUE TITLE
[Feature] Add support for programs that allow you to create disk RAIDs

### ISSUE THREAD

**Issue by kerta1n** [IceWhaleTech/CasaOS#1024]
In order for casaOS to gain an edge over OSes like Unraid and TrueNAS, it would be very useful and extremely cool to see casaOS help manage multiple disks at the same time (and have the ability to create drive pools, allowing for encryption with different filesystems such as btrFS, xfs, zfs, etc), while at the same time, use the packages available from the user's running distro.

**Comment by ETWang1991** [comment]
Understanding your needs, if you had to choose between the above unraid and turenas and only one necessary core feature, what feature would you want?
Frankly, for CasaOS, we're looking for streamlining, and we'd like to figure out your ideal way to organize your disks before we do

**Comment by kerta1n** [comment]
I would prefer an unRAID approach, as it's Parity and high water configurations allow easier switching of drives. I have run unRAID myself and can assist in studying how the disks are configured if needed.

**Comment by bmentink** [comment]
I prefer MergerFS + SnapRAID approach ... if you google around, a lot of users prefer SnapRAID over unRAID.

**Comment by aniel300** [comment]
> I prefer MergerFS + SnapRAID approach ... if you google around, a lot of users prefer SnapRAID over unRAID.

snapraid is not real time

**Comment by aniel300** [comment]
any chances of this project ever implementing raid and tiered cache system/ solution

**Comment by ETWang1991** [comment]
Raid not in six months, manpower does not support this @aniel300

**Comment by kerta1n** [comment]
@ETWang1991 also another reason to avoid snapraid in this use case, it can be difficult to reconfigure, and it's primarily targeted at those who are only storing media files

> SnapRAID is mainly targeted for a home media center, with a lot of big files that rarely change.

Those who edit large files frequently may deal with performance issues, so it should be noted for future development to develop a solution similar to what unRAID has done/cache system that @aniel300 mentioned.

**Comment by buff0k** [comment]
Another approach would be to implement allowing you add a local folder as "storage", that way an imported zpool (zfs) would be usable as a storage location at least?

**Comment by KRDucky** [comment]
ZFS hands down. Makes it very simple to move my drives around. Reinstalling the OS and re-adding my pool is a snap. Data integrity is top notch. I used to run mdraid and lvm2 arrays but when I went ZFS, I have not looked back. I have unraid on another server and it irritates me a lot. Especially since if I want to do basically anything change wise, I have to stop the docker containers etc just to add or remove a drive. VERY irritating.

**Comment by k5adb** [comment]
I have to agree with KRDucky ZFS is what I've been using for a long time on my HP N40l servers at home.

**Comment by KRDucky** [comment]
> > I prefer MergerFS + SnapRAID approach ... if you google around, a lot of users prefer SnapRAID over unRAID.
> 
> snapraid is not real time

neither is unraid. ZFS is realtime, fast, stable and reliable. It also has built in functionality for caching.

**Comment by aniel300** [comment]
unraid if i remember correctly is real time

**Comment by aniel300** [comment]
> > > I prefer MergerFS + SnapRAID approach ... if you google around, a lot of users prefer SnapRAID over unRAID.
> > 
> > 
> > snapraid is not real time
> 
> neither is unraid. ZFS is realtime, fast, stable and reliable. It also has built in functionality for caching.

unraid if i remember correctly is real time
